carapace-server/src/main/java/org/carapaceproxy/utils/CarapaceLogger.java

```diff
@@ -41,7 +42,15 @@ public static void setLoggingDebugEnabled(boolean loggingDebugEnabled) {
         CarapaceLogger.loggingDebugEnabled = loggingDebugEnabled;
     }
 
-    public static void debug(String s, Object ... o) {
-        LOG.log(loggingDebugEnabled ? Level.INFO : Level.FINE, s, o);
+    /**
+     * This method assumes that {@link Level#DEBUG debug log level} is disabled.
+     * If {@link #setLoggingDebugEnabled(boolean) Carapace log level was enabled},
+     * forwards the log event to the {@link Level#INFO info log level} instead of using the debug level.
+     *
+     * @param s the format string
+     * @param o a list of arguments
+     */
+    public static void debug(String s, Object... o) {
+        (loggingDebugEnabled ? LOG.atInfo() : LOG.atDebug()).log(s, o);
```

Comment:
I don't like this tricky approach at all, but I couldn't find any better approach that wouldn't break current `logging.debug.enabled` configuration, so I kept this approach.

If we ever think about revamping the configuration, I think we should drop this class and just leverage logback configuration (or the native configuration of whatever backend we will choose to put in our classpath)

Reply:
Take a look at this link to the Logback docs: <https://logback.qos.ch/manual/configuration.html#autoScan>

Is "Automatically reloading configuration file upon modification" what you are looking for?

Reply:
@NiccoMlt Yes this is what i mean...it's already implemented?

Reply:
Yes, kinda; this is a built-in Logback feature.

Logback's auto-scan feature automatically detects changes to the logging configuration file and reloads the configuration without restarting the application. If the configuration file (e.g., `logback.xml`) is modified, Logback automatically reloads it. If the configuration file contains errors during the reload, Logback keeps the current configuration.

The point is, to enable it we need to include the `scan` attribute in the `<configuration>` element, and we are currently **not** doing so, but I certainly enable it; then, when we update `logback.xml` **in the classpath** we get it auto-reloaded.

This obviously means that if we launch a self-contained Carapace JAR, we cannot edit it (because the resource is inside the archive). A possible workaround would be to use an external configuration file with the `logback.configurationFile` property.

Reply:
Done!

- dropped the `logging.debug.enabled` property, replacing the logs with debugs
- enabled `scan` on the default configuration